### PR TITLE
Don't rely on members to query if syncing user can post to room

### PR DESCRIPTION
**Pull request by bwindels**
Required by https://github.com/matrix-org/matrix-react-sdk/pull/2145

### CODE REVIEW

File: src/models/room-state.js [hunk 1]
```diff
@@ -663,25 +677,29 @@ RoomState.prototype._maySendEventOfType = function(eventType, userId, state) {
 
     let state_default = 0;
     let events_default = 0;
+
     if (power_levels_event) {
         power_levels = power_levels_event.getContent();
         events_levels = power_levels.events || {};
 
-        if (utils.isNumber(power_levels.state_default)) {
+        if (Number.isFinite(power_levels.state_default)) {
```
Comment (ara4n) [inline src/models/room-state.js]: why the change?

**Comment by ara4n** [comment]
lgtm modulo one random q :)

File: src/models/room-state.js (hunk 1 above)
Comment (bwindels) [reply src/models/room-state.js]: Better to use things that are now available in the standard JS as opposed to rolling our own?

Comment (dbkr) [reply src/models/room-state.js]: There is another `utils.isNumber` below we should probably change for consistency

**Comment by bwindels** [comment]
making changes to avoid looking up the member, don't merge yet.

**Comment by bwindels** [comment]
ready for another look
